app/scripts/metamask-controller.js

```diff
@@ -254,6 +266,22 @@ module.exports = class MetamaskController {
     this.configManager.useEtherscanProvider()
     extension.runtime.reload()
   }
+
+  buyEth (address, amount) {
+    if (!amount) amount = '5'
+
+    var network = this.idStore._currentState.network
+    var url = `https://buy.coinbase.com/?code=9ec56d01-7e81-5017-930c-513daa27bb6a&amount=${amount}&address=${address}&crypto_currency=ETH`
+
+    if (network === '2') {
+      url = 'https://testfaucet.metamask.io/'
+    }
+
+    chrome.tabs.create({
```

Comment:
the reason its in metamask controller is original plan was to listen fo coinbases postmessage but testing for that was such a bitch that that was abandoned i didnt move back in hopes that i or we could get it working in the future